tests/test_cmdLineFwk.py

```diff
@@ -1158,6 +1158,34 @@ def testSimpleQGraphDatastoreRecords(self):
             else:
                 self.assertEqual(quantum.datastore_records, {})
 
+    def testSummary(self):
+        """Test generating a summary report."""
+        args = _makeArgs(butler_config=self.root, input="test", output="output")
+        butler = makeSimpleButler(self.root, run=args.input, inMemory=False)
+        populateButler(self.pipeline, butler)
+
+        fwk = CmdLineFwk()
+        taskFactory = AddTaskFactoryMock()
+
+        qgraph = fwk.makeGraph(self.pipeline, args)
+        self.assertEqual(len(qgraph.taskGraph), self.nQuanta)
+        self.assertEqual(len(qgraph), self.nQuanta)
+
+        # Ensure that the output run used in the graph is also used in
+        # the pipeline execution. It is possible for makeGraph and runPipeline
+        # to calculate time-stamped runs across a second boundary.
+        args.output_run = qgraph.metadata["output_run"]
+
+        with makeTmpFile(suffix=".json") as tmpname:
```

Comment:
I had not noticed that there was a special temp file context manager written for this test file. Next time I'm looking I need to replace that with something more standard. The content parameter is only used in one place so it doesn't seem like we are gaining much.